src/components/elements/table/GenericTable.tsx

```diff
@@ -71,6 +72,7 @@ export interface Props<T> {
   selectable?: 'row' | 'checkbox'; // selectable by clicking row or by clicking checkbox
   selected?: readonly string[]; // selection can optionally be controlled by the parent
   isRowSelectable?: (row: T) => boolean;
+  getRowSelectDisabledReason?: (row: T) => string | undefined;
```

Comment:
Could these be combined into one function? They seem closely linked, in that `getRowSelectDisabledReason` would only ever be passed if `isRowSelectable` is also passed, and they would have the same internal logic for determining whether the row is selectable or not.


```suggestion
  isRowSelectable?: (row: T) => {
    selectable: boolean; // Whether this specific row can be selected (if `selectable` prop is passed)
    reason?: string; // Reason why this row cannot be selected (displayed in tooltip)
  };
```

Reply:
I will take a stab at cleaning this up. I think I kept them separate since a disabled reason is not always passed, but maybe it would be good to require.

Reply:
Looks like we can excise this for now anyway based on https://github.com/greenriver/hmis-frontend/pull/1037#discussion_r1950857673